### DISCUSSION TITLE
Swiper angular not working when rendered inside ng-content of a generic component

### DISCUSSION THREAD

**Discussion by paulmartinjoy** [nolimits4web/swiper#5380] [Q&A]
I am trying to create a generic component which is intended to make the styling in the project similar. Each page has image slider as well as content data. When i try to do that, the swiper is not functional. An example to make it clear: 

The following is the generic styling component which has the requisite styling: 

    <div class="wrapper">
      <div class="wrapper-content">
        <ng-content select=".images"> </ng-content>
      </div>  
    </div>

It is used in another page like:

    <app-generic-component>
      <div class="images">
        <swiper [config]="config">
          <ng-template swiperSlide *ngFor="let img of images">
           <img [src]="img"/>       
          </ng-template>
        </swiper>
      </div>
    </app-generic-component>

Swiper does not work as it should. It works perfectly fine if it is not wrapped inside the generic component. If someone knows how to make it working, it would be really helpful.

**Comment by vltansky** [comment]
codesandbox / stackblitz would be very helpful.

**Reply by julian-viirtue** [reply]
Could you solve this? I am trying to replicate the same in angular

**Comment by nvahalik** [comment]
Having the same issue.

We are coming from Ionic 5/6 and were using [`ion-slides`](https://github.com/ionic-team/ionic-framework/blob/2cba460ef682b0f570e02be3a486b3db04f77b24/core/src/components/slides/slides.tsx). 

That component didn't [initialize swiper](https://github.com/ionic-team/ionic-framework/blob/2cba460ef682b0f570e02be3a486b3db04f77b24/core/src/components/slides/slides.tsx#L341-L352) until _after_ all the [slide content had loaded](https://github.com/ionic-team/ionic-framework/blob/2cba460ef682b0f570e02be3a486b3db04f77b24/core/src/components/slides/slides.tsx#L518-L522).

It seems like in this case, the only real option here is to implement something with virtual slides using `@ContentChildren` and wiring up those templates to be used by `ng-template`s within the component.
